etna/models/tbats.py

```diff
@@ -156,7 +172,10 @@ def predict_components(self, df: pd.DataFrame) -> pd.DataFrame:
             raise ValueError("Model is not fitted! Fit the model before estimating forecast components!")
 
         if self._last_train_timestamp < df["timestamp"].max() or self._first_train_timestamp > df["timestamp"].min():
-            raise ValueError("To estimate out-of-sample prediction decomposition use `forecast` method.")
+            raise NotImplementedError(
```

Comment:
I think it is a reasonable way to go. 

Do we want to do this in this task? Or it is better to do this in some other task? Because it can require more planning than changing error messages like we do in this task.

Reply:
You can make it in another task if some research is needed